src/js/elements/subindicator_filter/filter_controller.js

```diff
@@ -0,0 +1,223 @@
+    }
+
+    removeFilterRow(filterRow) {
+        this._filterRows = this._filterRows.filter(filter => filter != filterRow)
+        this.triggerEvent(FilterControllerModel.EVENTS.filterRemoved, filterRow)
+    }
+
+    reset() {
+        this._filterRows.forEach(filterRow => {
+            this.triggerEvent(FilterControllerModel.EVENTS.filterRemoved, filterRow)
+        })
+
+        this._filterRows = []
+    }
+
+    setDataFilterModel(dataFilterModel) {
+        this.reset();
+
+        this._dataFilterModel = dataFilterModel;
+        this.triggerEvent(FilterControllerModel.EVENTS.dataFilterModelUpdated, this);
+    }
+}
+
+export class FilterController extends Observable {
+
+    static EVENTS = {
+        'ready': 'FilterController.ready'
+    }
+
+    /*
+    Note to self
+
+    About to figure out how to filter the data appropriately and send it via signal for the choropleth to be created
+    After that, evaluate how easy it is to adapt this to the charts
+
+    Things that don't currently work:
+    1. Disabling the add filter button isn't working great - it checks the number of available filters, not the number of filterRowson the screen
+    2. Removing filterRows is a little messsy
+    */
+
+    constructor(container) {
+        super();
+
+        this._container = container;
+        this._model = new FilterControllerModel();
+        this._addFilterButton = new AddFilterButton(); // TODO should pass in a container
+        this._dataFilterModel = null;
+
+        this.prepareDomElements();
+        this.prepareEvents();
+    }
+
+    get container() {
+        return this._container;
+    }
+
+    get model() {
+        return this._model;
+    }
+
+    get addFilterButton() {
+        return this._addFilterButton;
+    }
+
+    prepareDomElements() {
+        this._rowContainer = $(this.container).find(filterRowClass)[0];
+        $(this._rowContainer).hide();
+        // $(this.container).find(filterRowClass).hide();
+    }
+
+    prepareEvents() {
+        const self = this;
+        this.addFilterButton.on(AddFilterButton.EVENTS.clicked, () => {
+            self.addEmptyFilter()
+        })
+    }
+
+    addInitialFilterRow(dataFilterModel) {
+        // let rowContainer = $(this.container).find(filterRowClass)[0]
+        // $(rowContainer).show();
+
+        let isDefault = true;
+        let isExtra = false;
+
+        // this.model.filterRows.push(new FilterRow(rowContainer, dataFilterModel, isDefault, isExtra))
+        this.addEmptyFilter(isDefault, isExtra);
+    }
+
+    addEmptyFilter(isDefault = false, isExtra = true) {
+        if (this.model.dataFilterModel.availableFilters.length > 0) {
+            const self = this;
+
+            let filterRowContainer = this._rowContainer.cloneNode(true);
+            $(filterRowContainer).show();
+
+            let filterRow = new FilterRow(filterRowContainer, this.model.dataFilterModel, isDefault, isExtra);
+            this.model.filterRows.push(filterRow);
```

Comment:
I think instead of pushing directly to the array, we should create an `addFilter` method on the model